src/Controller/InvoiceController.php

```diff
@@ -170,6 +171,28 @@ public function edit(Request $request, Invoice $invoice, InvoiceRepository $invo
         ]);
     }
 
+    #[Route('/{id}/generate-description', name: 'app_invoices_generate_description', methods: ['GET'])]
+    public function generateDescription(Invoice $invoice, $defaultInvoiceDescriptionTemplate): JsonResponse
+    {
+        $description = $defaultInvoiceDescriptionTemplate;
+
+        // Default description.
+        if (!empty($invoice->getClient())) {
+            $projectLeadName = $invoice->getClient()?->getProjectLeadName() ?? null;
+            $projectLeadMail = $invoice->getClient()?->getProjectLeadMail() ?? null;
```

Comment:
You know that `$invoice->getClient()` is set inside the `if` block (under the assumption that `getClient` always returns the same value), but a rewrite to 

```suggestion
        $client = $invoice->getClient();
        if (!empty($client)) {
            $projectLeadName = $client->getProjectLeadName() ?? null;
            $projectLeadMail = $client->getProjectLeadMail() ?? null;
```

may be an improvement.